### PR TITLE
Rotation invariant brief

### PR DESCRIPTION
**Pull request by GilLevi**
Added rotation invariance to the BRIEF descriptor. The approach is explained and evaluated in the following my blog post:

https://gilscvblog.wordpress.com/2015/01/02/adding-rotation-invariance-to-the-brief-descriptor/

The descriptor itself is implemented in opencv_contrib (I'll make a separate pull request), in the opencv repository I only added a tutorial.

### CODE REVIEW

**Comment by StevenPuttemans** [comment]
Hi @GilLevi! Basically you used TABS for indentation and the buildbot cannot handle that. You need to replace each tab by 3 spaces. Then the documentation bot will run just fine!

**Comment by GilLevi** [comment]
Thanks, @StevenPuttemans ! I'll fix that.

File: samples/cpp/tutorial_code/xfeatures2D/RI_BRIEF_match.cpp [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+using namespace std;
+using namespace cv;
+
+const float inlier_threshold = 2.5f; // Distance threshold to identify inliers
+const float nn_match_ratio = 0.8f;   // Nearest neighbor matching ratio
+
+int main(void)
+{
+	Mat img1 = imread("D:/OpenCV3.0GitHub/opencv/samples/data/graf1.png", IMREAD_GRAYSCALE);
```
Comment (ilya-lavrenov) [inline samples/cpp/tutorial_code/xfeatures2D/RI_BRIEF_match.cpp]: please, don't use absolute names

**Comment by mshabunin** [comment]
@GilLevi, since the different solution has already been integrated in Itseez/opencv_contrib#207, I think this PR can be considered obsolete.

I'd suggest to transform this to sample application in [xfeatures2d/samples](https://github.com/Itseez/opencv_contrib/tree/master/modules/xfeatures2d/samples) folder or to tutorial (_xfeatures2d/tutorials_) + sample application (_xfeatures2d/samples_).

**Pull request closed without merging**